pep-0755.rst

```diff
@@ -0,0 +1,551 @@
+by those library maintainers that decide to opt-in into this feature.
+
+Any authors that decide not to adopt it, are free to continue using docstrings
+with any particular format they decide, no docstrings at all, etc.
+
+This argument could be analogous to the argument against type annotations
+in general, as they do indeed increase verbosity, in exchange for their
+features. But again, as with type annotations, this would be optional and only
+to be used by those that are willing to take the extra verbosity in exchange
+for the benefits.
+
+
+Doc is not Typing
+-----------------
+
+It could also be argued that documentation is not really part of typing, or that
+it should live in a different module. Or that this information should not be part
+of the signature but live in another place (like the docstring).
+
+Nevertheless, type annotations in Python could already be considered, by default,
+mainly documentation: they carry additional information about variables,
+parameters, return types, and by default they don't have any runtime behavior.
+
+It could be argued that this proposal extends the type of information that
+type annotations carry, the same way as :pep:`702` extends them to include
+deprecation information.
+
+And as described above, including this in ``typing_extensions`` to support older
+versions of Python would have a very simple and practical benefit.
+
+
+Multiple Standards
+------------------
+
+Another argument against this would be that it would create another standard,
+and that there are already several pseudo-standards for docstrings. It could
+seem better to formalize one of the currently existing standards.
+
+Nevertheless, as stated above, none of those standards cover the general
+drawbacks of a doctsring-based approach that this proposal solves naturally.
+
+None of the editors have full docstring editing support (even when they have
+rendering support). Again, this is solved by this proposal just by using
+standard Python syntax and structures instead of a docstring microsyntax.
+
+The effort required to implement support for this proposal by tools would
+be minimal compared to that required for alternative docstring-based
+pseudo-standards, as for this proposal, editors would only need to
+access an already existing value in their ASTs, instead of writing a parser
+for a new string microsyntax.
+
+In the same way, it can be seen that, in many cases, a new standard that
+takes advantage of new features and solves several problems from previous
+methods can be worth having. As is the case with the new ``pyproject.toml``,
+``dataclass_transform``, the new typing pipe/union (``|``) operator, and other cases.
+
+
+Adoption
+--------
+
+As this is a new standard proposal, it would only make sense if it had
+interest from the community.
+
+Fortunately there's already interest from several mainstream libraries
+from several developers and teams, including FastAPI, Typer, SQLModel,
```

Comment:
It might also be good to get buy-in from Sphinx (cc @AA-Turner) or other documentation tools. I imagine Sphinx would use this PEP's syntax for rendering documentation.

Reply:
Thanks! @AA-Turner what do you think? Of course, this is an opt-in thing, not enforced to developers.

But do you think it would be viable to support this in Sphinx in some way? Is there something I could do to help in this area?